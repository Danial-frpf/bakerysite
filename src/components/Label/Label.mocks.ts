import { ILabel } from ".";

const small: ILabel = {
    label: "Small",
    size: "sm",
};

const medium: ILabel = {
    label: "Medium",
    size: "md",
};

const large: ILabel = {
    label: "Large",
    size: "lg",
};

const extraLarge: ILabel = {
    label: "ExtraLarge",
    size: "xl",
};

export const mockLabelProps = {
    small,
    medium,
    large,
    extraLarge,
};
