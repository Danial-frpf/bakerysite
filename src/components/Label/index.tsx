export interface ILabel {
    label: string;
    size?: "sm" | "md" | "lg" | "xl";
    props?: React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLHeadingElement>,
        HTMLHeadingElement
    >;
}
const Label: React.FC<ILabel> = ({ label, size = "md", props }) => {
    if (size === "sm")
        return (
            <p data-testid="Label" {...props}>
                <span>{label}</span>
            </p>
        );

    if (size === "md")
        return (
            <h4 data-testid="Label" {...props}>
                <span>{label}</span>
            </h4>
        );

    if (size === "lg")
        return (
            <h2 data-testid="Label" {...props}>
                <span>{label}</span>
            </h2>
        );

    return (
        <h1 data-testid="Label" {...props}>
            <span>{label}</span>
        </h1>
    );
};

export default Label;
