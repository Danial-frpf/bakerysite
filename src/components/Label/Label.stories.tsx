import { ComponentStory, ComponentMeta } from "@storybook/react";
import Label, { ILabel } from ".";
import { mockLabelProps } from "./Label.mocks";

export default {
    title: "templates/Label",
    component: Label,
    // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
    argTypes: {},
} as ComponentMeta<typeof Label>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Label> = (args) => <Label {...args} />;

export const Small = Template.bind({});
export const Medium = Template.bind({});
export const Large = Template.bind({});
export const ExtraLarge = Template.bind({});

// More on args: https://storybook.js.org/docs/react/writing-stories/args

Small.args = {
    ...mockLabelProps.small,
} as ILabel;

Medium.args = {
    ...mockLabelProps.medium,
} as ILabel;

Large.args = {
    ...mockLabelProps.large,
} as ILabel;

ExtraLarge.args = {
    ...mockLabelProps.extraLarge,
} as ILabel;
