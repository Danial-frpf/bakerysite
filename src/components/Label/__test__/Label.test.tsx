import { createRoot } from "react-dom/client";
import { act } from "react-dom/test-utils";
import Label from "..";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(<Label label="test"></Label>);
    });
});

it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(<Label label="test" size="sm"></Label>);
    });
});

it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(<Label label="test" size="lg"></Label>);
    });
});

it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(<Label label="test" size="xl"></Label>);
    });
});

it("renders Label correctly", () => {
    const { getByTestId } = render(<Label label="test"></Label>);
    expect(getByTestId("Label")).toHaveTextContent("test");
});

it("renders Label correctly", () => {
    const { getByTestId } = render(<Label label="test" size="sm"></Label>);
    expect(getByTestId("Label")).toHaveTextContent("test");
});

it("renders Label correctly", () => {
    const { getByTestId } = render(<Label label="test" size="lg"></Label>);
    expect(getByTestId("Label")).toHaveTextContent("test");
});

it("renders Label correctly", () => {
    const { getByTestId } = render(<Label label="test" size="xl"></Label>);
    expect(getByTestId("Label")).toHaveTextContent("test");
});

it("matches snapshot", () => {
    const tree = renderer.create(<Label label="test"></Label>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<Label label="test" size="sm"></Label>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<Label label="test" size="lg"></Label>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<Label label="test" size="xl"></Label>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
