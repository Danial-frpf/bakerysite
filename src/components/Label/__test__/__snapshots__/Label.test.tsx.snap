// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h4
  data-testid="Label"
>
  <span>
    test
  </span>
</h4>
`;

exports[`matches snapshot 2`] = `
<p
  data-testid="Label"
>
  <span>
    test
  </span>
</p>
`;

exports[`matches snapshot 3`] = `
<h2
  data-testid="Label"
>
  <span>
    test
  </span>
</h2>
`;

exports[`matches snapshot 4`] = `
<h1
  data-testid="Label"
>
  <span>
    test
  </span>
</h1>
`;
